src/argilla_sdk/settings/_vector.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, TYPE_CHECKING
+
+from argilla_sdk._api._vectors import VectorsAPI
+from argilla_sdk._models import VectorFieldModel
+from argilla_sdk._resource import Resource
+from argilla_sdk.client import Argilla
+
+if TYPE_CHECKING:
+    from argilla_sdk import Dataset
+
+__all__ = ["VectorField"]
+
+
+class VectorField(Resource):
```

Comment:
Vectors (and metadata) definition differs from Fields and Question. This is why I decided to remove the inheritance from the `SettingsPropertyBase`. Otherwise, it creates some extra weird methods just to keep aligned with the interface.